auth: zsh completion script for pdnsutil

tokred opened the pull request:
### Short description
Zsh completion script for pdnsutil, incl. completion of subcommands, zones and tsig keys (bare with me, my first completion script ever)

### Checklist
<!-- please indicate if any of these things are done/included with this Pull Request. Not all boxes need to be checked for the Pull Request to be accepted -->
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

Habbie commented on contrib/_pdnsutil.zsh_completion:
```diff
@@ -0,0 +1,305 @@
+# put file _pdnsutil somewhere into $fpath,
+#       e.g. /usr/share/zsh/vendor-completions/
+#
+# commands based on Debian Bullseye package pdns-server 4.4
```
Can you update this to 4.6? It would be a pity to merge something that is immediately out of date.

tokred replied:
Thank you, that's a fair point. I have updated the code to reflect pdnsutil 4.6 (and added minor corrections to handle edge cases if no tsig keys or no zones are available yet).

As mentioned, my first completion ever but hope it improves usability of pdnsutil :)

tachitworapong commented:
> ### Short description
> Zsh completion script for pdnsutil, incl. completion of subcommands, zones and tsig keys (bare with me, my first completion script ever)
> 
> ### Checklist
> I have:
> 
> * [x]  read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
> * [ ]  compiled this code
> * [x]  tested this code
> * [ ]  included documentation (including possible behaviour changes)
> * [x]  documented the code
> * [ ]  added or modified regression test(s)
> * [ ]  added or modified unit test(s)

tachitworapong commented:
> ### Short description
> Zsh completion script for pdnsutil, incl. completion of subcommands, zones and tsig keys (bare with me, my first completion script ever)
> 
> ### Checklist
> I have:
> 
> * [x]  read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
> * [ ]  compiled this code
> * [x]  tested this code
> * [ ]  included documentation (including possible behaviour changes)
> * [x]  documented the code
> * [ ]  added or modified regression test(s)
> * [ ]  added or modified unit test(s)

tokred commented:
@Habbie, is there anything else left do to from my side? Kindly let me know.

Regards

Habbie commented:
Hi! Sorry, this one slipped my attention. Meanwhile a few new commands have appeared. I believe this list is complete (I hope I got the format right):

```
'remove-autoprimary:Remove an autoprimary'
'list-autoprimaries:List all autoprimaries'
'export-zone-key-pem:Export to stdout in PEM the private key described'
'import-zone-key-pem:Import a private key from a PEM file'
'zonemd-verify-file:Validate ZONEMD for ZONE'
```

if you add those, I'll try to merge soon. After that, responsibility for keeping the completions updated lies with whoever adds commands :)

tokred commented:
`remove-autoprimary` and `list-autoprimaries` were already included in the file, but I added `import/export-zone-key-pem` and `zonemd-verify-file` to the completion, as requested. Squashed my commits and force-pushed a new version.

Regards

Habbie commented:
Thank you!
